### ISSUE TITLE
Tile VM <> Django backend integration

### ISSUE THREAD

**Issue by kgeographer** [WorldHistoricalGazetteer/whg3#94]
I told the bot my mental model of how this might go, and it read it back to me in the inimitable bot outline style:

**Django Signals for Dataset and Collection Models**:

- Utilizing Django signals to trigger actions when a dataset or collection is flagged is a good strategy. This automates the process and integrates it seamlessly with your existing Django application.
- The signal can trigger a process that communicates (e.g., via SSH or a REST API call) with your tile generation VM. This process will include sending the necessary data and the specific Tippecanoe command to generate the tiles.

**Tile Generation Command Execution:**

- The command sent to the tile VM should specify the data to be used, the Tippecanoe specifications (like zoom levels, detail levels, etc.), and the target location for storing the generated tiles.
- Ensure that the VM has the necessary permissions and capabilities to receive and execute these commands securely.

**Asynchronous Task Management and Notifications:**

- Integrating Celery for handling asynchronous tasks is a sound approach. It can manage the task of communicating with the tile VM, monitoring the progress, and handling the completion message.
- When the tile generation process is initiated, you can send a notification to the user (or to admin/editorial staff) indicating that the map is being processed. This enhances the user experience by setting the right expectations.
- 

**Completion Notification and Availability:**

- Once the tile generation is complete, the system can notify the relevant parties. This could be implemented through a callback mechanism or a polling system where the Django server periodically checks the status of the tile generation process.
- Celery or another task queue can manage this checking mechanism, updating the status of the process in your Django application.

**Front-End Integration:**

- On the client side, pages that will render the maps can check the availability of the tiles. If the tiles are not yet ready, you can display a placeholder message like "Map under construction" or a similar notification.
- Once the tiles are available, the page dynamically updates to render the map using a client-side library like MapLibre GL.

**Security and Error Handling:**

- Ensure that the communication between your Django server and the tile VM is secure, especially if you're transferring data or commands over SSH or another protocol.
- Implement robust error handling to manage scenarios where tile generation fails or is delayed.

This workflow provides a clear, automated process from the moment a dataset is flagged for map generation to the point where interactive maps are available on your site. It leverages the strengths of Django for web application management, Celery for task queuing and asynchronous operations, and external tile generation tools, creating a cohesive and efficient system.
